hazelcast-client/src/main/java/com/hazelcast/client/impl/clientside/FailoverClientConfigSupport.java

```diff
@@ -38,41 +43,85 @@ private FailoverClientConfigSupport() {
-     * If still could no create HazelcastException is thrown.
+     * Loads failover configuration using the search logic described in
+     * {@link HazelcastClient#newHazelcastFailoverClient()} and builds
+     * {@link ClientFailoverConfig}.
      *
-     * @param clientFailoverConfig provided via HazelcastClient.newHazelcastClient(ClientFailoverConfig config)
      * @return resolvedConfigs
+     * @throws HazelcastException            if no failover configuration is found
      * @throws InvalidConfigurationException when given config is not valid
      */
-    public static ClientFailoverConfig resolveClientConfig(ClientFailoverConfig clientFailoverConfig) {
+    public static ClientFailoverConfig resolveClientFailoverConfig() {
+        return resolveClientFailoverConfig(locateAndCreateClientFailoverConfig());
+    }
+
+    /**
+     * If {@code clientFailoverConfig} passed {@code null}, then loads
+     * failover configuration using the search logic described in
+     * {@link HazelcastClient#newHazelcastFailoverClient()} and builds
+     * {@link ClientFailoverConfig}.
+     *
+     * @param clientFailoverConfig provided via {@link HazelcastClient#newHazelcastFailoverClient(ClientFailoverConfig)}
+     * @return resolvedConfigs
+     * @throws HazelcastException  if {@code clientFailoverConfig} is {@code null} and no failover configuration is found
+     * @throws InvalidConfigurationException when given config is not valid
+     */
+    public static ClientFailoverConfig resolveClientFailoverConfig(ClientFailoverConfig clientFailoverConfig) {
         if (clientFailoverConfig == null) {
-            XmlClientFailoverConfigBuilder configBuilder = new XmlClientFailoverConfigBuilder();
-            clientFailoverConfig = configBuilder.build();
+            clientFailoverConfig = locateAndCreateClientFailoverConfig();
         }
         checkValidAlternative(clientFailoverConfig.getClientConfigs());
         return clientFailoverConfig;
     }
 
     /**
-     * Returns a ClientFailoverConfig with single client config. If clientConfig is null,
-     * We create it via XmlClientConfigBuilder().build()
+     * Returns a {@link ClientFailoverConfig} with single client config.
+     * If the provided {@code clientConfig} is {@code null}, client
+     * configuration is loaded using the resolution logic as described in
+     * {@link HazelcastClient#newHazelcastClient()}
      *
-     * @param config provided via HazelcastClient.newHazelcastClient(ClientConfig config)
+     * @param config provided via {@link HazelcastClient#newHazelcastClient(ClientConfig)}
      * @return resolvedConfigs
      * @throws InvalidConfigurationException when given config is not valid
      */
     public static ClientFailoverConfig resolveClientConfig(ClientConfig config) {
         if (config == null) {
-            config = createDefaultClientConfig();
+            config = locateAndCreateClientConfig();
```

Comment:
I did not notice that this is trying to create a failover config before. 
My expectation is that when a user calls `HazelcastClient#newHazelcastClient(ClientConfig)`, we should not look for failover config from system property or classpath. This is what we documented also on HazelcastClient API.

They should be used only when `newHazelcast